sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -650,24 +613,24 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
             try
             {
                 int sessionIndex = 0;
-                string sessionId = null;
+                SessionLifeCycleManager sessionLifeCycle = null;
                 while (!cancellationToken.IsCancellationRequested)
                 {
                     await MessageHandlerSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
                     // hold onto all the tasks that we are starting so that when cancellation is requested,
                     // we can await them to make sure we surface any unexpected exceptions, i.e. exceptions
                     // other than TaskCanceledExceptions
 
-                    if (IsSessionProcessor && _sessionIds.Length > 0)
+                    if (_sessionLifeCycles.Count > 0)
```

Comment:
Why do you need this condition?

Reply:
Can you just use foreach here?

Reply:
I'm not looping though. I just want to get the next sessionLifeCycle, and I keep track of the index within the while loop.

Reply:
Actually, we wouldn't enter the loop for non-session case, so it wouldn't get tasks added.